src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1405 @@
+    :param ignore: Unix shell style pattern to ignore when searching for files,
+        defaults to None
+    :param dataset_list_uri: URI with a list of dataset URIs, defaults to None
+    :param max_files: maximum number of URIs to read/find,
+        defaults to None (no limit)
+    :param compression_filter: serialized tiledb filter,
+        defaults to None
+    :param workers: number of workers for dataset ingestion, defaults to MAX_WORKERS
+    :param batch_size: batch size for dataset ingestion, defaults to BATCH_SIZE
+    :param tile_size: for rasters this is the tile (block) size
+        for the merged destination array, defaults to 1024
+    :param pixels_per_fragment: for rasters this this the number of tiles
+        that will be processed together per fragment. Ideally aim to align
+        as a factor of tile_size
+    :param chunk_size: for point cloud this is the PDAL chunk size, defaults to 1000000
+    :param nodata: NODATA value for raster merging
+    :param resampling: string, resampling method,
+        one of None, bilinear, cubic, nearest and average
+    :param res: Tuple[float, float], output resolution in x/y
+    :param stats: bool, print TileDB stats to stdout
+    :param verbose: verbose logging, defaults to False
+    :param trace: bool, enabling log tracing, defaults to False
+    :param log_uri: log array URI
+    """
+
+    logger = get_logger_wrapper(verbose)
+
+    if log_uri:
+        create_log_array(log_uri)
+
+    funcs = {
+        DatasetType.POINTCLOUD: {
+            "udf_fn": ingest_point_cloud_udf,
+        },
+        DatasetType.RASTER: {
+            "udf_fn": ingest_raster_udf,
+        },
+        DatasetType.GEOMETRY: {
+            "udf_fn": ingest_geometry_udf,
+        },
+    }
+
+    fn = funcs[dataset_type]["udf_fn"]
+
+    # Build the task graph
+    dag_name = f"{dataset_type.name}-{DEFAULT_DAG_NAME}"
+    task_prefix = f"{dataset_type.name} - Task"
+
+    logger.info("Building graph")
+    graph = tiledb.cloud.dag.DAG(
+        name=dag_name,
+        mode=tiledb.cloud.dag.Mode.BATCH,
+        max_workers=workers,
+        namespace=namespace,
+    )
+
+    # find sources
+    input_list_node = graph.submit(
+        build_inputs_udf,
+        dataset_type=dataset_type,
+        access_credentials_name=acn,
+        config=config,
+        search_uri=search_uri,
+        pattern=pattern,
+        ignore=ignore,
+        dataset_list_uri=dataset_list_uri,
+        max_files=max_files,
+        compression_filter=compression_filter,
+        tile_size=tile_size,
+        pixels_per_fragment=PIXELS_PER_FRAGMENT,
+        chunk_size=chunk_size,
+        nodata=nodata,
+        resampling=resampling,
+        res=res,
+        verbose=verbose,
+        trace=trace,
+        log_uri=log_uri,
+        name=f"{dag_name} input collector",
+    )
+
+    # schema creation node, returns a sequence of work items
+    ingest_node = graph.submit(
+        fn,
+        **input_list_node,
```

Comment:
`input_list_node` is a `Node` type, you can't deconstruct the dictionary that is returned from `input_list_node` like this. The dictionary needs to be passed in-intact and handles inside the functions.

Reply:
👍